src/dp.rs

```diff
@@ -0,0 +1,62 @@
+// SPDX-License-Identifier: MPL-2.0
+
+//! Differential privacy (DP) primitives.
+use crate::field::FieldElement;
+use crate::vdaf::prg::SeedStream;
+use std::fmt::Debug;
+
+/// Positive rational number to represent DP parameters and noise distributions.
+#[allow(dead_code)]
+pub struct Rational {
+    numerator: u32,
+    denominator: u32,
+}
+
+impl Rational {
+    #[allow(dead_code)]
+    fn new(numerator: u32, denominator: u32) -> Self {
+        Rational {
+            numerator,
+            denominator,
+        }
+    }
+}
+/// Trait for a noise distribution over a (vector space over a) finite field.
+/// Useful for differential privacy.
+/// Inherited by specific mechanism instantiations such as `[DiscreteGaussian]`
+/// that will enforce the VDAF-dependent DP guarantees.
+pub trait Distribution: Clone + Debug {
+    /// Fills the `noise` buffer with elements from the distribution over `F`,
+    /// with randomness coming from `seed_stream`
+    fn sample<F: FieldElement, S: SeedStream>(&self, seed_stream: S, noise: &mut [F]) -> Vec<F>;
```

Comment:
`seed_stream` should be a `&mut S`, otherwise we can't use the `seed_stream` again after calling `sample`